src/Components/Filter/Filter.tsx

```diff
@@ -9,23 +11,36 @@ interface Props {
 export const Filter: FC<Props> = ({ genresList }) => {
   const [genres, setGenres] = useState(genresList);
 
-  const genreClickHandler = (id: number): void => {
+  const dispatch = useAppDispatch();
+
+  const genreClickHandler = (name: string): void => {
     const updatedGenres = genres.map((genre) => {
-      genre.active = genre.id === id;
+      genre.active = genre.name === name;
 
       return genre;
     });
 
     setGenres([...updatedGenres]);
   };
 
+  useEffect(() => {
+    dispatch(
+      setFilter(
+        genres
```

Comment:
I would separate this into a separate variable before the dispatch to reduce chain

Reply:
Thanks.